perf/sawtooth_perf/src/batch_gen.rs

```diff
@@ -130,36 +130,82 @@ impl<'a> SignedBatchProducer<'a> {
             signer: signer,
         }
     }
+}
+
+impl<'a> Iterator for SignedBatchProducer<'a> {
+
+    type Item = BatchResult;
 
     /// Gets the next BatchResult.
     /// `Ok(None)` indicates that the underlying source has been consumed.
-    pub fn next_batch(&mut self) -> BatchResult {
+    fn next(&mut self) -> Option<BatchResult> {
         let txns = match self.transaction_source.next(self.max_batch_size) {
             Ok(txns) => txns,
-            Err(err) => return Err(BatchingError::MessageError(err)),
+            Err(err) => return Some(Err(BatchingError::MessageError(err))),
         };
 
-        if txns.len() == 0 {
-            return Ok(None);
-        }
+        batch_transactions(txns, self.signer)
+    }
+}
 
-        let mut batch_header = BatchHeader::new();
+fn batch_transactions(txns: Vec<Transaction>, signer: &signing::Signer) -> Option<BatchResult> {
+    if txns.len() == 0 {
```

Comment:
Why should this be treated as invalid? Within the context of this function, it seems like passing a zero length transaction vector should be valid and checking the length should be done outside the function? Adding this check makes the return type more complex.